core/primitives/src/views.rs

```diff
@@ -32,31 +34,56 @@ use crate::types::{
 /// A view of the account
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
 pub struct AccountView {
+    /// A spendable balance (before state rent charge)
     #[serde(with = "u128_dec_format")]
     pub amount: Balance,
+    /// An actual balance in storage
+    #[serde(with = "u128_dec_format")]
+    pub actual_amount: Balance,
+    /// The amount locked due to staking
     #[serde(with = "u128_dec_format")]
     pub locked: Balance,
+    /// Hash of the code stored in the storage for this account.
     pub code_hash: CryptoHash,
+    /// Storage used by the given account (in bytes).
     pub storage_usage: StorageUsage,
+    /// Last height at which the storage was paid for.
     pub storage_paid_at: BlockHeight,
 }
 
-impl From<Account> for AccountView {
-    fn from(account: Account) -> Self {
+impl AccountView {
+    /// Calculates spendable amount and transforms into the AccountView
+    pub fn from_account(
+        account_id: &AccountId,
+        account: &Account,
+        block_height: BlockHeight,
+        account_length_baseline_cost_per_block: Balance,
+        storage_cost_byte_per_block: Balance,
+    ) -> AccountView {
+        let actual_amount = account.amount;
+        let amount = actual_amount
+            - calculate_rent(
```

Comment:
Let's calculate rent with the full amount and use `saturating_sub` here

Reply:
I changed the AccountView to:

```rust
/// A view of the account
#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
pub struct AccountView {
    /// A balance from storage
    #[serde(with = "u128_dec_format")]
    pub amount: Balance,
    /// Storage rent to pay on next transaction
    #[serde(with = "u128_dec_format")]
    pub rent: Balance,
    /// The amount locked due to staking
    #[serde(with = "u128_dec_format")]
    pub locked: Balance,
    /// Hash of the code stored in the storage for this account.
    pub code_hash: CryptoHash,
    /// Storage used by the given account (in bytes).
    pub storage_usage: StorageUsage,
    /// Last height at which the storage was paid for.
    pub storage_paid_at: BlockHeight,
}
```
I think it's better to separate values and allow to calculate the spendable amount on the `nearlib` side.

Reply:
Perfect